src/goto-analyzer/static_verifier.cpp

```diff
@@ -27,6 +27,39 @@ struct static_verifier_resultt
   irep_idt function_id;
 };
 
+void static_verifier(
+  const abstract_goto_modelt &abstract_goto_model,
+  const ai_baset &ai,
+  propertiest &properties)
+{
+  auto const ns = namespacet{abstract_goto_model.get_symbol_table()};
+  for(auto &property : properties)
+  {
+    auto &property_status = property.second.status;
+    auto const &property_location = property.second.pc;
+    auto condition = property_location->get_condition();
+    auto const predecessor = ai.abstract_state_before(property_location);
+    auto const &domain = *predecessor;
+    domain.ai_simplify(condition, ns);
+    if(condition.is_true())
+    {
+      property_status = property_statust::PASS;
+    }
+    else if(condition.is_false())
+    {
+      property_status = property_statust::FAIL;
```

Comment:
Is this a definite fail? If not, then it should be `UNKNOWN`.

Reply:
Yes, if a condition is false that means that using domain information it could be simplified to false, i.e. definite fail.